components/lib/splitter/Splitter.js

```diff
@@ -15,11 +15,15 @@ export const Splitter = React.memo(
         const prevPanelElement = React.useRef(null);
         const nextPanelElement = React.useRef(null);
         const prevPanelSize = React.useRef(null);
+        const prevPanelSizeNew = React.useRef(null);
         const nextPanelSize = React.useRef(null);
+        const nextPanelSizeNew = React.useRef(null);
         const prevPanelIndex = React.useRef(null);
-        const panelSizes = React.useRef(null);
-        const mounted = React.useRef(false);
+        const [panelSizes, setPanelSizes] = React.useState([]);
         const isStateful = props.stateKey != null;
+        const childrenLength = props.children && props.children.length;
```

Comment:
Hmmm won't childrenLength be a `boolean` here and not an integer?  Shouldn't it be:

```js
const childrenLength = props.children ? props.children.length : 0;
```

won't dividing by 0 below return `Infinity` ?

Reply:
It is more of a safeguard against `Cannot read properties of undefined`. I do not see a risk of division by zero, because panelSize function will never be invoked if there are zero children.